pkg/local_object_storage/shard/put.go

```diff
@@ -41,19 +41,18 @@ func (s *Shard) Put(prm *PutPrm) (*PutRes, error) {
 	)
 
 	if s.hasWriteCache() {
-		res, err = s.writeCache.Put(putPrm)
-		if err != nil {
-			s.log.Debug("can't put message to writeCache, trying to blobStor")
-
-			res = nil // just in case
+		err = s.writeCache.Put(prm.obj)
+		if err == nil {
+			return nil, nil
 		}
+
+		s.log.Debug("can't put message to writeCache, trying to blobStor")
+		res = nil // just in case
```

Comment:
You don't need this assignment anymore. Also remove comment about `// res == nil if there is no writeCache..`